roles/openstack-setup/tasks/images.yml

```diff
@@ -1,7 +1,12 @@
 ---
+# NOTE: Glance does not know how to proxy copy-from URLs, so we should instead
+# download the image files locally and create images using the file method.
+- name: download images locally
+  get_url: url={{ item.url }} dest=/tmp/{{ item.filename }}
+  with_items: glance.images
```

Comment:
Yeah, we probably should drop a file somewhere that notes that we've uploaded the image. Or just check the API to see if the image exists.